### PR TITLE
Extract SSL configuration handling to a separate class

### PR DESCRIPTION
**Pull request by mccheah**
More or less a drag-and-drop of all the SSL related logic.

Part of #109

### CODE REVIEW

**Comment by mccheah** [comment]
Soft dependency on #122 to fix merge conflicts as the refactors are done in a chain.

**Comment by mccheah** [comment]
@ash211 @foxish @iyanuobidele this had a soft dependency on #122 which is now merged - can we look to review and merge this one now?

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala [hunk 1]
```diff
@@ -688,8 +563,8 @@ private[spark] class Client(
   private def buildDriverSubmissionClient(
       kubernetesClient: KubernetesClient,
       service: Service,
-      driverSubmitSslOptions: SSLOptions): KubernetesSparkRestApi = {
-    val urlScheme = if (driverSubmitSslOptions.enabled) {
+      kubernetesSslConfiguration: KubernetesSslConfiguration): KubernetesSparkRestApi = {
```
Comment (foxish) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: nit: can we just call it SSLConfiguration? The prefix `Kubernetes` seems to be everywhere.

**Comment by iyanuobidele** [comment]
LGTM

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesSslConfigurationProvider.scala [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  }
+
+  private def parseDriverSubmitSslOptions(): (SSLOptions, Boolean) = {
+    val maybeKeyStore = sparkConf.get(KUBERNETES_DRIVER_SUBMIT_KEYSTORE)
+    val resolvedSparkConf = sparkConf.clone()
+    val (isLocalKeyStore, resolvedKeyStore) = maybeKeyStore.map(keyStore => {
+      val keyStoreURI = Utils.resolveURI(keyStore)
+      val isProvidedKeyStoreLocal = keyStoreURI.getScheme match {
+        case "file" | null => true
+        case "container" => false
```
Comment (ash211) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesSslConfigurationProvider.scala]: I think container isn't supposed to work anymore after this change https://github.com/apache-spark-on-k8s/spark/pull/107/files#diff-b5527f236b253e0d9f5db5164bdb43e9R79

**Review by ash211 — CHANGES_REQUESTED** [review]

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala (hunk 1 above)
Comment (mccheah) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala]: Done

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesSslConfigurationProvider.scala (hunk 2 above)
Comment (mccheah) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesSslConfigurationProvider.scala]: Done - changed to `local`

**Review by ash211 — APPROVED** [review]
LGTM let's merge when the build is green

**Pull request merged**